test/common/upstream/upstream_impl_test.cc

```diff
@@ -128,11 +128,48 @@ TEST_P(StrictDnsParamTest, ImmediateResolve) {
   NiceMock<MockClusterManager> cm;
   StrictDnsClusterImpl cluster(parseClusterFromJson(json), runtime, stats, ssl_context_manager,
                                dns_resolver, cm, dispatcher, false);
-  cluster.setInitializedCb([&]() -> void { initialized.ready(); });
+  cluster.initialize([&]() -> void { initialized.ready(); });
   EXPECT_EQ(2UL, cluster.hosts().size());
   EXPECT_EQ(2UL, cluster.healthyHosts().size());
 }
 
+// Resolve zero hosts, while using health checking.
+TEST(StrictDnsClusterImplTest, ZeroHostsHealthChecker) {
+  Stats::IsolatedStoreImpl stats;
+  Ssl::MockContextManager ssl_context_manager;
+  auto dns_resolver = std::make_shared<Network::MockDnsResolver>();
+  NiceMock<Event::MockDispatcher> dispatcher;
+  NiceMock<Runtime::MockLoader> runtime;
+  NiceMock<MockClusterManager> cm;
+  ReadyWatcher initialized;
+
+  const std::string json = R"EOF(
+  {
+    "name": "name",
+    "connect_timeout_ms": 250,
+    "type": "strict_dns",
+    "lb_type": "round_robin",
+    "hosts": [{"url": "tcp://foo.bar.com:443"}]
+  }
+  )EOF";
```

Comment:
Is it time to start writing new unit tests v2 proto?

Reply:
Sure I can give it a go.